### PR TITLE
Remove usage of WaitAndGetResult

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@dotnet/roslyn-ide @JoeRobich @jasonmalinowski ptal.

File: src/EditorFeatures/Core/InlineRename/InlineRenameSession.cs [hunk 1]
```diff
@@ -588,7 +589,8 @@ private void QueueApplyReplacements()
         return (replacementInfo, mergeResult);
     }
 
-    private void ApplyReplacements(IInlineRenameReplacementInfo replacementInfo, LinkedFileMergeSessionResult mergeResult, CancellationToken cancellationToken)
+    private async Task ApplyReplacementsAsync(
+        IInlineRenameReplacementInfo replacementInfo, LinkedFileMergeSessionResult mergeResult, CancellationToken cancellationToken)
     {
         _threadingContext.ThrowIfNotOnUIThread();
```
Comment (sharwell) [inline src/EditorFeatures/Core/InlineRename/InlineRenameSession.cs:595]: This should be changed to `await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync` (which will be a NOP but will better fit expected patterns with the return type change)

**Review by sharwell — APPROVED** [review]

File: src/EditorFeatures/Core/InlineRename/InlineRenameSession.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/EditorFeatures/Core/InlineRename/InlineRenameSession.cs:595]: Will do.

Comment (CyrusNajmabadi) [reply src/EditorFeatures/Core/InlineRename/InlineRenameSession.cs]: done.

**Pull request merged**